src/components/Pagination/Pagination.jsx

```diff
@@ -1,163 +1,220 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect, useReducer } from 'react';
 import PropTypes from 'prop-types';
 
-import { Flex, PseudoBox, useTheme } from '@chakra-ui/core';
+import { Flex, Button, useTheme } from '@chakra-ui/core';
 
 const LEFT_PAGE = 'LEFT';
 const RIGHT_PAGE = 'RIGHT';
+const SET_PAGINATION_CONFIG = 'SET_PAGINATION_CONFIG';
+const SET_CURRENT_PAGE = 'SET_CURRENT_PAGE';
+const SET_PAGE_NUMBERS = 'SET_PAGE_NUMBERS';
+
+/**
+ * @function range
+ *
+ * @param {number} from - Range start value
+ * @param {number} to - Range end value
+ * @param {number} step - Number to increment each loop count by
+ * @returns {Array.<number>} - An array of numbers
+ */
+
+const range = (from, to, step = 1) => {
+  let i = from;
+  const range = [];
+
+  for (let i = from; i <= to; i = i + step) {
+    range.push(i);
+  }
+
+  return range;
+};
 
 /**
  * @function Pagination
  *
  * @param {number} totalRecords - Total records count
- * @param {function} onPageChanged - Callback function that passes paginated information
  * @param {number} pageLimit - Number of items to display per page
  * @param {number} pageNeighbors - Number of neighboring pages when pagination is in the middle of the list
+ * @param {function} onPageChanged - Callback function that passes paginated information
  *
  */
 
+function reducer(state, action) {
+  switch (action.type) {
+    case SET_PAGINATION_CONFIG:
+      return {
+        ...state,
+        ...action.payload,
+      };
+    case SET_CURRENT_PAGE:
+      return {
+        ...state,
+        currentPage: action.payload,
+      };
+    case SET_PAGE_NUMBERS: {
+      return {
+        ...state,
+        pages: action.payload,
+      };
+    }
+    default:
+      return state;
+  }
+}
+
 const Pagination = props => {
   const theme = useTheme();
 
-  const [totalRecords, setTotalRecords] = useState(null);
-  const [totalPages, setTotalPages] = useState(0);
-  const [pageLimit, setPageLimit] = useState(10);
-  const [pageNeighbors, setPageNeighbors] = useState(1);
-  const [currentPage, setCurrentPage] = useState(1);
-  const [pages, setPages] = useState([]);
+  const [state, dispatch] = useReducer(reducer, {
+    totalRecords: null,
+    totalPages: 0,
+    pageLimit: 0,
+    pageNeighbors: 1,
+    currentPage: 1,
+    pages: [],
+  });
 
   useEffect(() => {
-    const { totalRecords, pageLimit, pageNeighbors } = props;
-
-    if (pageLimit) {
-      setPageLimit(pageLimit);
-    }
-    if (pageNeighbors) {
-      setPageNeighbors(Math.max(0, Math.min(pageNeighbors, 2)));
-    }
+    const {
+      currentPage = 1,
+      totalRecords = 0,
+      pageLimit = 10,
+      pageNeighbors = 1,
+    } = props;
```

Comment:
You need to destructure `props` in the component's parameter list or at the very beginning to avoid getting an error saying you haven't declared `props` as a dependency.

Reply:
When you say beginning, do you mean at the near the top where this function component is declared?